internal/kubernetes/drainer.go

```diff
@@ -91,6 +91,13 @@ type APICordonDrainer struct {
 	evictionHeadroom time.Duration
 }
 
+// SuppliedCondition defines the condition will be watched.
+type SuppliedCondition struct {
+	Type               core.NodeConditionType
+	Status             core.ConditionStatus
+	MinimumWaitingTime int64
```

Comment:
Based on some of the other feedback to make this a `time.Duration`, could we rename this to `MinimumDuration`.